packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/categories/CreateCategoriesTests.java

```diff
@@ -0,0 +1,223 @@
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+
+import java.util.List;
+
+import org.alfresco.rest.RestTest;
+import org.alfresco.rest.model.RestCategoryModel;
+import org.alfresco.rest.model.RestCategoryModelsCollection;
+import org.alfresco.utility.data.RandomData;
+import org.alfresco.utility.model.FolderModel;
+import org.alfresco.utility.model.SiteModel;
+import org.alfresco.utility.model.TestGroup;
+import org.alfresco.utility.model.UserModel;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+public class CreateCategoriesTests extends RestTest
+{
+    private static final String FIELD_NAME = "name";
+    private static final String FIELD_PARENT_ID = "parentId";
+    private static final String FIELD_HAS_CHILDREN = "hasChildren";
+    private static final String FIELD_ID = "id";
+    private UserModel user;
+
+    @BeforeClass(alwaysRun = true)
+    public void dataPreparation() throws Exception
+    {
+        STEP("Create a user");
+        user = dataUser.createRandomTestUser();
+    }
+
+    /**
+     * Check we can create a category as direct child of root category
+     */
+    @Test(groups = {TestGroup.REST_API})
+    public void testCreateCategoryUnderRoot()
+    {
+        STEP("Create a category under root category (as admin)");
+        final RestCategoryModel rootCategory = new RestCategoryModel();
+        rootCategory.setId("-root-");
+        final RestCategoryModel aCategory = new RestCategoryModel();
+        aCategory.setName(RandomData.getRandomName("Category"));
+        final RestCategoryModel createdCategory = restClient.authenticateUser(dataUser.getAdminUser())
+                .withCoreAPI()
+                .usingCategory(rootCategory)
+                .createSingleCategory(aCategory);
+        restClient.assertStatusCodeIs(CREATED);
+
+        createdCategory.assertThat()
+                .field(FIELD_NAME).is(aCategory.getName());
+        createdCategory.assertThat()
+                .field(FIELD_PARENT_ID).is(rootCategory.getId());
+        createdCategory.assertThat()
+                .field(FIELD_HAS_CHILDREN).is(false);
+    }
+
+    /**
+     * Check we can create several categories as children of a created category
+     */
+    @Test(groups = {TestGroup.REST_API})
+    public void testCreateSeveralSubCategories()
+    {
+        STEP("Create a category under root category (as admin)");
+        final RestCategoryModel rootCategory = new RestCategoryModel();
+        rootCategory.setId("-root-");
+        final RestCategoryModel aCategory = new RestCategoryModel();
+        aCategory.setName(RandomData.getRandomName("Category"));
+        final RestCategoryModel createdCategory = restClient.authenticateUser(dataUser.getAdminUser())
+                .withCoreAPI()
+                .usingCategory(rootCategory)
+                .createSingleCategory(aCategory);
+        restClient.assertStatusCodeIs(CREATED);
+
+        createdCategory.assertThat()
```

Comment:
I'm not sure if you have to check all the fields here as it was tested in the first test method (`testCreateCategoryUnderRoot()`). In context of this test only the `FIELD_HAS_CHILDREN` is important.